### PR TITLE
Unique terms get all

### PR DESCRIPTION
**Pull request by dionboles-asym**
this adds get_all to unique_terms and fixes progressbar counts

### CODE REVIEW

File: cdapython/results/factories/collect_result.py [hunk 1]
```diff
@@ -58,7 +59,17 @@ def get_all(
         output: str = "",
         page_size: Union[None, int] = None,
         show_bar: bool = True,
-    ) -> Result:
+    ):
+        """
+        This will page
```
Comment (ahaessly) [inline cdapython/results/factories/collect_result.py:64]: did you mean to say more in this description? "This will page"...

File: cdapython/utils/utility.py [hunk 2]
```diff
@@ -140,7 +146,9 @@ def unique_terms(
 
     if async_req is None:
         async_req = False
-    col_name = col_name
+
+    if limit is not None:
```
Comment (ahaessly) [inline cdapython/utils/utility.py:150]: what if the limit is still a very large number where we would still want to pull the results in pages?

**Review by ahaessly — COMMENTED** [review]
just a few questions

File: cdapython/__init__.py [hunk 3]
```diff
@@ -112,11 +113,12 @@
     import builtins as __builtin__
 
 
-def console_print(*args: Any, **kwargs: Any) -> None:
-    from rich.console import Console
-
-    console: Console = Console()
-    console.print(*args, **kwargs)
+def console_print(*args, **kwargs) -> None:
+    if "file" in kwargs:
```
[the remark this answers is not in the document]
Comment (dionboles-asym) [reply cdapython/__init__.py:117]: yes i can

File: cdapython/utils/utility.py (hunk 2 above)
Comment (dionboles-asym) [reply cdapython/utils/utility.py:150]: This right now is here as a hold over for users that use limit instead of page_size but i don't know if i should autopage for people over 100

**Review by ahaessly — APPROVED** [review]
👍

File: cdapython/__init__.py [hunk 4]
```diff
@@ -112,11 +113,18 @@
...
 
 
-def console_print(*args: Any, **kwargs: Any) -> None:
-    from rich.console import Console
-
-    console: Console = Console()
-    console.print(*args, **kwargs)
+def console_print(*args, **kwargs) -> None:
+    """
+    This This function is used to override the default python `print` method.
```
Comment (otchet-broad) [inline cdapython/__init__.py:118]:
```suggestion
    This function is used to override the default python `print` method.
```

File: cdapython/results/factories/collect_result.py [hunk 5]
```diff
@@ -46,7 +47,7 @@ def __init__(
 
     def extend_result(self, result: Result) -> None:
         """_summary_
-        This will method will
+        This will method will concat the list value to a global list
```
Comment (otchet-broad) [inline cdapython/results/factories/collect_result.py:50]:
```suggestion
        This method will concat the list value to a global list
```

File: cdapython/results/factories/collect_result.py [hunk 6]
```diff
@@ -59,6 +60,17 @@ def get_all(
         page_size: Union[None, int] = None,
         show_bar: bool = True,
     ) -> Result:
+        """
+        This will page through the query results automatically looping and appending result objects to a List
+        for the user.
+        Args:
+            output (str, optional): if you add this output string "full_list"or"full_df" You will either get a list or a DataFrame. Defaults to "".
+            page_size (int, optional): this can increase the page size results . Defaults to None.
```
Comment (otchet-broad) [inline cdapython/results/factories/collect_result.py:68]:
```suggestion
            page_size (int, optional): this can increase the page size results . Defaults to None.  Known issue: the first page is always 100, even when the size is adjusted.
```

File: cdapython/results/factories/collect_result.py [hunk 7]
```diff
@@ -68,8 +80,27 @@ def paginator(
         to_list: bool = False,
         page_size: int = None,
         show_bar: bool = False,
-    ):
-        return super().paginator(output, to_df, to_list, page_size, show_bar)
+    ) -> Paginator:
+        """
+        This will return a pagination class which is used in iteration objects, such as the for loop this will.
```
Comment (otchet-broad) [inline cdapython/results/factories/collect_result.py:85]:
```suggestion
        This will return a pagination class (Collection) for iteration, such as the for loop.
```

File: cdapython/results/factories/collect_result.py [hunk 8]
```diff
@@ -68,8 +80,27 @@ def paginator(
...
+        By default, the pagination object will return a result class, which is normally what you will get from a `Q.run`
+        Args:
+            output (str, optional): if you add this output string "full_list"or"full_df" You will either get a list or a DataFrame. Defaults to "".
+            to_df (bool, optional): This will return a list from the DataFrame. Defaults to False.
+            to_list (bool, optional): This will return a list from the paginator. Defaults to False.
+            page_size (int, optional): this can increase the page size results . Defaults to None.
```
Comment (otchet-broad) [inline cdapython/results/factories/collect_result.py:91]:
```suggestion
            page_size (int, optional): this can change the page size of results . Defaults to results limit.
```

File: cdapython/utils/utility.py [hunk 9]
```diff
@@ -104,31 +105,36 @@ def unique_terms(
...
-    files: Optional[bool] = False,
     show_sql: bool = False,
     show_counts: bool = False,
     verbose: bool = True,
+    limit: Optional[int] = None,
 ) -> Union[Result, StringResult, ColumnsResult, None]:
-    """[summary]
-
+    """
+    This will return unique terms Result object.
```
Comment (otchet-broad) [inline cdapython/utils/utility.py:120]:
```suggestion
    This will return a unique terms Result object.
```

File: docker-compose.yml [hunk 10]
```diff
@@ -4,7 +4,7 @@ services:
     env_file:
       - .env
     container_name: 'cdapython-notebook'
-    image: jupyter/minimal-notebook:latest
+    image: jupyter/minimal-notebook:2023-05-29
```
Comment (otchet-broad) [inline docker-compose.yml:7]: Why is this changed to a specific version?

File: poetry.lock [hunk 11]
```diff
@@ -660,7 +682,7 @@ files = [
 name = "exceptiongroup"
 version = "1.1.1"
 description = "Backport of PEP 654 (exception groups)"
-category = "dev"
+category = "main"
```
Comment (otchet-broad) [inline poetry.lock:685]: Why did this change?

Comment (otchet-broad) [reply poetry.lock:685]: AnyIO

**Review by otchet-broad — APPROVED** [review]
Nice work and discussion.  Left some minor feedback to clean up some of the comments.

File: docker-compose.yml (hunk 10 above)
Comment (dionboles-asym) [reply docker-compose.yml:7]: This change was made because  the latest image uses python 3.11. Currently this branch dose not support 3.11 but a future branch will.
